### PR TITLE
python37Packages.py3status: Fix prefetch replacements

### PR DESCRIPTION
**Pull request by guserav**
###### Motivation for this change
I found my py3status crashing with ```py3status: setup error (FileNotFoundError) parse_config.py line 722``` after doing an update. It seems like this error comes from the fact that the replaces of in this case file to the path in /nix/store are failing because the py3status repo changed from '' to ""

###### Things done
Adjusted the sed commands and checked with the code base if they would apply.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/py3status/default.nix [hunk 1]
```diff
@@ -32,18 +30,11 @@ buildPythonPackage rec {
 
   doCheck = false;
   propagatedBuildInputs = [ pytz requests tzlocal i3ipc pydbus pygobject3 pyserial ];
-  buildInputs = [ file ];
-  prePatch = ''
-    sed -i -e "s|'file|'${file}/bin/file|" py3status/parse_config.py
-    sed -i -e "s|\[\"acpi\"|\[\"${acpi}/bin/acpi\"|" py3status/modules/battery_level.py
-    sed -i -e "s|notify-send|${libnotify}/bin/notify-send|" py3status/modules/battery_level.py
-    sed -i -e "s|/usr/bin/whoami|${coreutils}/bin/whoami|" py3status/modules/external_script.py
```
Comment (JohnAZoidberg) [inline pkgs/development/python-modules/py3status/default.nix]: This must still be patched, right?

Comment (JohnAZoidberg) [reply pkgs/development/python-modules/py3status/default.nix]: Well, it's only mentioned in a comment.

**Comment by JohnAZoidberg** [comment]
Please rebase on top of latest master.

**Comment by jonringer** [comment]
unfortunately makeWrapper doesn't compose with python packages, as someone else can import the package, and now the package is still calling out to `/usr/bin/whoami` and what not.

The previous inline patching is considered best practice, the package probably just needs to be updated as it has new paths which need to be patched

**Comment by guserav** [comment]
@jonringer Thank you for the comment.
With that knowledge I belief the PR is obsolete.

**Comment by jonringer** [comment]
OTOH, if this package is only meant to be used as an application. Then we can move it out of python-modules and make it it's own application, then using wrapping mechanisms are alright because it should be isolated at the application level, and you don't have to worry about it being imported by other python packages.

**Comment by jonringer** [comment]
@guserav this is a leaf package (not being used by other python packages), so if you want, you can reopen this PR, but just make py3status an entry in `all-packages.nix` and move it out of python-modules.

**Pull request closed without merging**